File: p-hero/twoSum.js


// const nums = [2,7,11,15]
// const target = 9;
const nums = [3,3]
const target = 6;
// const nums = [3,2,4]
// const target = 6;

const twoSum = (nums, target) => {
  for(let i=0; i<nums.length; i++){
    for(let j=i+1; j<nums.length; j++)
    {
      if(nums[i] + nums[j] === target)
      {
        return [i, j]
      }
    }
  }
}

console.log(twoSum(nums, target))

File: practice/two_sum.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
  vector<int> nums = {3, 2, 4};
  int target = 6;

  unordered_map<int, int> mp;

  for(int i =0; i<nums.size(); i++){
    int complement = target - nums[i];

    if(mp.find(complement) != mp.end()){
      cout << mp[complement] <<"," << i;
      return 0;
    }

    mp[nums[i]] = i;
  }



  return 0; 
}